internal/debug/flags.go

```diff
@@ -241,27 +212,55 @@ func Setup(ctx *cli.Context) error {
 		} else {
 			context = append(context, "location", filepath.Join(os.TempDir(), "geth-lumberjack.log"))
 		}
-		lumberWriter := &lumberjack.Logger{
+		fileOutput = &lumberjack.Logger{
 			Filename:   logFile,
 			MaxSize:    ctx.Int(logMaxSizeMBsFlag.Name),
 			MaxBackups: ctx.Int(logMaxBackupsFlag.Name),
 			MaxAge:     ctx.Int(logMaxAgeFlag.Name),
 			Compress:   ctx.Bool(logCompressFlag.Name),
 		}
-		ostream = log.StreamHandler(io.MultiWriter(output, lumberWriter), logfmt)
+		output = io.MultiWriter(terminalOutput, fileOutput)
 	} else if logFile != "" {
-		f, err := os.OpenFile(logFile, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
-		if err != nil {
+		var err error
+		if fileOutput, err = os.OpenFile(logFile, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644); err != nil {
```

Comment:
the lumberjack.Logger is an `io.Closer`, and so is the `fileOutput`. So in both these cases, you need to set the `logOutputIF` to it